src/widgets/DataTable/DataTableHeaderRow.js

```diff
@@ -22,37 +22,40 @@ import { newDataTableStyles } from '../../globalStyles';
  *
  *
  */
-const DataTableHeaderRow = React.memo(({ columns, sortBy, isAscending, dispatch, sortAction }) => (
-  <HeaderRow
-    style={newDataTableStyles.headerRow}
-    renderCells={() =>
-      columns.map(({ key, title, sortable, width, alignText }, index) => {
-        const sortDirection = isAscending ? 'ASC' : 'DESC';
-        const directionForThisColumn = key === sortBy ? sortDirection : null;
-        const isLastCell = index === columns.length - 1;
-        const { headerCells, cellText } = newDataTableStyles;
-        return (
-          <HeaderCell
-            key={key}
-            title={title}
-            SortAscComponent={SortAscIcon}
-            SortDescComponent={SortDescIcon}
-            SortNeutralComponent={SortNeutralIcon}
-            columnKey={key}
-            onPressAction={sortable ? sortAction : null}
-            dispatch={dispatch}
-            sortDirection={directionForThisColumn}
-            sortable={sortable}
-            width={width}
-            containerStyle={headerCells[alignText || 'left']}
-            textStyle={cellText[alignText || 'left']}
-            isLastCell={isLastCell}
-          />
-        );
-      })
-    }
-  />
-));
+const DataTableHeaderRow = React.memo(({ columns, sortBy, isAscending, dispatch, sortAction }) => {
+  const { headerRow, headerCells, cellText } = dataTableStyles;
+  return (
+    <HeaderRow
+      style={headerRow}
+      renderCells={() =>
+        columns.map(({ key, title, sortable, width, alignText }, index) => {
+          const sortDirection = isAscending ? 'ASC' : 'DESC';
+          const directionForThisColumn = key === sortBy ? sortDirection : null;
+          const isLastCell = index === columns.length - 1;
```

Comment:
Here I do think you want to use `>=` rather than `===`. Acknowledging what you said about preferring errors last time, I think here it's not guaranteed that you'll throw an error if you're off by one (e.g. if somehow you skip the last cell and ran over), so it's probably a good idea to play defensively.

Reply:
we're mapping over columns, if one was skipped and run over, we'd be in a different array and probably be seg faulting or something..!

Reply:
Similar to the defensive programming point above. There's no harm in not protecting against it, so IMO it makes sense to. Could be a hangover from C++ days, but in general it seems like a bad idea to rely on assumptions (IMO most bugs are the result of unforeseen results of assumptions).

Reply:
![defensive](https://user-images.githubusercontent.com/43223637/65468569-9a902f00-deb8-11e9-88b0-7deb433ef219.png)